### PR TITLE
enh(ecmascript): Add built-in types

### LINKED ISSUE #2992: (ecmascript) Missing BigInt type
**Describe the issue**
`BigInt` is not highlighted as type the same way `Number` is.
Actual: BitInt not recognized

**Which language seems to have the issue?**
`ecmascript` and therefore also `javascript`

**Sample Code to Reproduce**
```js
let bi = BigInt('1');
```

**Expected behavior**
Expected: BitInt recognized

**Additional context**
Also seems to be missing `BigInt64Array` and `BigUint64Array`.
See [List of built-ins](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects).

### PR DESCRIPTION
**Pull request by il3ven**
Resolves #2992 

### Changes
Added these built-in types **BigInt64Array, BigUint64Array, BigInt**
Created a new test file in javascript (built-in.txt)

### Checklist
- [x] Added markup tests, or they don't apply here because...
- [x] Updated the changelog at `CHANGES.md`
- [ ] Added myself to `AUTHORS.txt`, under Contributors

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -15,6 +15,7 @@ Language grammar improvements:
 - enh(php) Add `mixed` to list of keywords (#2997) [Ayesh][]
 - enh(php) Add support binary, octal, hex and scientific numerals with underscore separator support (#2997) [Ayesh][]
 - enh(php) Add support for Enums (#3004) [Ayesh][]
+- enh(ecmascript) Add built-in types [Vaibhav Chanana][]
```
Comment (joshgoebel) [inline CHANGES.md:18]: Please also finish this link by adding it's target on line ~25ish.

**Review by joshgoebel — APPROVED** [review]
Looks great, just finish up the changelog!

File: CHANGES.md (hunk 1 above)
Comment (il3ven) [reply CHANGES.md:18]: Done

Comment (il3ven) [reply CHANGES.md:18]: The link is already there down in the document. Do we need to add it again?

**Review by joshgoebel — APPROVED** [review]

File: CHANGES.md (hunk 1 above)
Comment (joshgoebel) [reply CHANGES.md:18]: For every release.  The document is split at release time - so only the current release exists.

**Pull request merged**